src/Symfony/Bundle/SecurityBundle/DependencyInjection/SecurityExtension.php

```diff
@@ -565,17 +564,14 @@ private function createAuthenticationListeners(ContainerBuilder $container, stri
                             throw new InvalidConfigurationException(sprintf('Cannot configure AuthenticatorManager as "%s" authentication does not support it, set "security.enable_authenticator_manager" to `false`.', $key));
                         }
 
-                        $authenticators = $factory->createAuthenticator($container, $id, $firewall[$key], $userProvider);
-                        if (\is_array($authenticators)) {
-                            foreach ($authenticators as $i => $authenticator) {
-                                $authenticationProviders[] = $authenticator;
-                            }
-                        } else {
-                            $authenticationProviders[] = $authenticators;
+                        $authenticators = (array) $factory->createAuthenticator($container, $id, $firewall[$key], $userProvider);
+                        foreach ($authenticators as $authenticator) {
+                            $authenticationProviders[] = $authenticator;
                         }
 
-                        if ($factory instanceof EntryPointFactoryInterface && ($entryPoint = $factory->registerEntryPoint($container, $id, $firewall[$key]))) {
-                            $entryPoints[$key] = $entryPoint;
+                        // allow entry point to be set to the authenticator key (e.g. "http_basic" or "form_login")
+                        if ($key === $defaultEntryPoint && \count($authenticators) === 1) {
+                            $defaultEntryPoint = $authenticators[0];
```

Comment:
todo: add a comment here to indicate that the new system uses a compiler pass to set the default entry point